src/app/dataset-view/additional-components/dataset-settings-component/dataset-settings.component.ts

```diff
@@ -95,6 +127,28 @@ export class DatasetSettingsComponent extends BaseComponent implements OnInit {
             });
     }
 
+    private initializationActiveTab(): void {
+        if (this.datasetPermissions.permissions.general.canSetVisibility) {
+            this.activeTab = this.getSectionFromUrl() ?? SettingsTabsEnum.GENERAL;
+        } else {
+            if (this.appConfigService.featureFlags.enableDatasetEnvVarsManagement) {
+                this.activeTab = SettingsTabsEnum.VARIABLES_AND_SECRETS;
+            } else {
+                promiseWithCatch(
+                    this.modalService.warning({
+                        message: "You don't have access to the variables and secrets manager",
+                        yesButtonText: "Ok",
+                    }),
+                );
+                this.navigationService.navigateToDatasetView({
+                    accountName: this.datasetBasics.owner.accountName,
+                    datasetName: this.datasetBasics.name,
+                    tab: DatasetViewTypeEnum.Overview,
+                });
+            }
+        }
+    }
```

Comment:
It looks like a routing guard should be doing this instead.
What is that modal warning even needed? Just get the user out of the page, if there is no permission.

Reply:
Discussed it in private chat. 
Came to the conclusion that it is necessary to implement routing for subtabs